### DISCUSSION TITLE
GPJ Mode creates 2nd DS3 Compatible HID Device, causing problems. Need help.

### DISCUSSION THREAD

**Discussion by goodstuff626** [nefarius/DsHidMini#216] [Support]
Using Windows 10, official ps3 controller, usb connection. Followed all instructions when installing. I've only used GPJ Gamepad plus Joystick & xinput device modes in DsHidMini. 

For some reason when I use GPJ mode it creates a 2nd DS3 Compatible HID Device in the windows game controllers control panel. When I properties test the top entry DS3 Compatible HID Device it has an extra z rotation, dial & slider, and 17 red buttons total. The second DS3 Compatible HID Device only shows the z axis, x rotation, y rotation, z rotation, slider & dial, and when I press buttons, it does not light up properly in the control panel. I cannot use GPJ mode with a lot of emulators because of the extra DS3 Compatible HID Device that gets created. 

When I use xinput it shows 1 DS3 Compatible HID, with a z axis, x rotation, y rotation, and 16 red buttons. This mode seems to work correctly. (is it supposed to show 16 red buttons?)

Please, how can I fix this problem?

**Comment by nefarius** [comment]
Fix what problem? That is intentional, the GPJ mode is literally there to give you **all the axes** in a DirectInput-compatible way, which is not possible on a single device because of DI's legacy axis limits. Quoted directly from the documentation: 

> Gamepad plus Joystick mode. In this mode, the controller is presented as two separate HID devices; a traditional gamepad featuring the same set of features like a common Xbox 360 controller and a Joystick, only handling the pressure axes.

**Reply by goodstuff626** [reply]
> Fix what problem? That is intentional, the GPJ mode is literally there to give you **all the axes** in a DirectInput-compatible way, which is not possible on a single device because of DI's legacy axis limits. Quoted directly from the documentation:
> 
> > Gamepad plus Joystick mode. In this mode, the controller is presented as two separate HID devices; a traditional gamepad featuring the same set of features like a common Xbox 360 controller and a Joystick, only handling the pressure axes.


But when I try to use GPJ mode with mame and duckstation, the input keys don't register correctly because of the second DS3 Compatible HID Device and it seems to cause all these alternate 2nd joystick inputs that makes every mapped button not work correct. Why can I only use xinput with mame and duckstation correctly? I would prefer to use GPJ mode to map all the buttons in any emulator. Why can't I use GPJ with mame & duckstation?

**Reply by Kanuan** [reply]
DirectInput compatible devices have a limit on how many axes/buttons they can have. The DS3 has so many axes/buttons because of its pressure sensitivity buttons that to fit them all you need 2 devices, which is what GPJ is for. One device has the normal gamepad functions and the other has only the pressure sensitivity axes related to the buttons.

Because you have the same controller being detected as 2 separated devices some tools won't be able to map them correctly, or you'll need to think outside of the box to workaround the conflicting inputs.

Although it is possible to have all buttons/axes of the DS3 in a single device, which is what SDF mode is for, it results in a non-compatible DirectInput device, meaning some axes may not be detected, or it may straight up cause issues depending on how the controller is being read.

The point I'm trying to make is the following: GPJ is working exactly as it was designed to. If it will work with the tools you want to use will depend on the tool itself and how you are trying to map it, and we can't offer further help on this second part.

Options to "mute" one or other part of the GPJ mode to facilitate mapping / prevent conflict could be included in DSHM configuration utility, but that may be over-engineering and Nef has the final say on everything around here.

**Reply by goodstuff626** [reply]
> DirectInput compatible devices have a limit on how many axes/buttons they can have. The DS3 has so many axes/buttons because of its pressure sensitivity buttons that to fit them all you need 2 devices, which is what GPJ is for. One device has the normal gamepad functions and the other has only the pressure sensitivity axes related to the buttons.
> 
> Because you have the same controller being detected as 2 separated devices some tools won't be able to map them correctly, or you'll need to think outside of the box to workaround the conflicting inputs.
> 
> Although it is possible to have all buttons/axes of the DS3 in a single device, which is what SDF mode is for, it results in a non-compatible DirectInput device, meaning some axes may not be detected, or it may straight up cause issues depending on how the controller is being read.
> 
> The point I'm trying to make is the following: GPJ is working exactly as it was designed to. If it will work with the tools you want to use will depend on the tool itself and how you are trying to map it, and we can't offer further help on this second part.
> 
> Options to "mute" one or other part of the GPJ mode to facilitate mapping / prevent conflict could be included in DSHM configuration utility, but that may be over-engineering and Nef has the final say on everything around here.

Ok, thanks for the info. Sorry, I'm not very good at this sort of thing. So I should use SDF mode? Does it map like dinput and I can map all buttons including ps3 home button? What is its drawback of "pressure sensitive axes"? Will that effect my ps3 controller in emulators? I only emulate ps2 and older.  And if I need pressure sensitive buttons, I could use xinput for that, correct?

**Reply by nefarius** [reply]
Duckstation works great with **SXS** mode, I literally set it up this week to play some Catlevania: Symphony of the Night, give that a try!
